src/app/backend/secrets.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+package main
+
+import (
+	api "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/labels"
+)
+
+// SecretSpec - a single secret representation
+type SecretSpec struct {
+	Name      string            `json:"name"`
+	Namespace string            `json:"namespace,omitempty"`
+	Type      api.SecretType    `json:"type,omitempty"`
+	Data      map[string][]byte `json:"data"`
+}
+
+// SecretsList - response structure for a queried secrets list
+type SecretsList struct {
+	// List of secrets
+	Secrets []SecretSpec `json:"secrets"`
+}
+
+// GetSecrets - return all secrets
+func GetSecrets(client *client.Client) (*SecretsList,
```

Comment:
This is only because I didn't want to update the secrets list via a request every time the namespace changes, due to performance concerns. But if the latency is negligible I will definitely change it :)